geopy/geocoders/mapbox.py

```diff
@@ -0,0 +1,209 @@
+        features = json['features']
+        if features == []:
+            return None
+
+        def parse_feature(feature):
+            location = feature['place_name']
+            place = feature['text']
+            longitude = feature['geometry']['coordinates'][0]
+            latitude = feature['geometry']['coordinates'][1]
+            return Location(location, (latitude, longitude), place)
+        if exactly_one:
+            return parse_feature(features[0])
+        else:
+            return [parse_feature(feature) for feature in features]
+
+    @staticmethod
+    def _coerce_point_to_string(point):
+        """
+        Do the right thing on "point" input. For geocoders with reverse
+        methods.
+        """
+        if isinstance(point, Point):
+            return ",".join((str(point.longitude), str(point.latitude)))
+        elif isinstance(point, (list, tuple)):
+            return ",".join((str(point[1]), str(point[2])))  # -altitude
+        elif isinstance(point, string_compare):
+            return ",".join(point.split(",")[::-1])
+        else:
+            raise ValueError("Invalid point")
+
+    def geocode(
+            self,
+            query,
+            exactly_one=True,
+            proximity=None,
+            country=None,
+            bbox=None,
+            timeout=DEFAULT_SENTINEL
+    ):
+        """
+        Return a location point by address
+
+        :param str query: The address or query you wish to geocode.
+
+        :param bool exactly_one: Return one result or a list of results, if
+            available.
+
+        :param proximity: a coordinate to bias local results based on a provided location.
+        :type proximity: list of tuple
+
+        :param string country: Country to filter result in form of
+            ISO 3166-1 alpha-2 country code.
+
+        :param bbox: The bounding box of the viewport within which
+            to bias geocode results more prominently.
+        :type bbox: list or tuple
+
+        :param int timeout: Time, in seconds, to wait for the geocoding service
+            to respond before raising a :class:`geopy.exc.GeocoderTimedOut`
+            exception. Set this only if you wish to override, on this call
+            only, the value set during the geocoder's initialization.
+        """
+        params = {}
+
+        params['access_token'] = self.api_key
+
+        if query is None or query == '':
+            raise GeocoderQueryError('query must be set')
+        if bbox:
+            if len(bbox) != 4:
+                raise GeocoderQueryError(
+                    "bbox must be a four-item iterable of lat,lon,lat,lon"
+                )
+            params['bbox'] = self._format_bbox_param(bbox)
+
+        if country:
+            if len(country) != 2:
+                raise GeocoderQueryError(
+                    "country must be specified as ISO 3166-1 alpha-2 country code."
+                )
```

Comment:
I would remove this check too. We don't really care about the length, and we cannot know for sure the supported languages anyway, to make the proper validation. So I'd let the backed validate this value.